network.go

```diff
@@ -1228,75 +1236,76 @@ func (n *network) updateSvcRecord(ep *endpoint, localEps []*endpoint, isAdd bool
 			ipv6 = iface.AddressIPv6().IP
 		}
 
+		serviceID := ep.svcID
```

Comment:
Since this method is a generic method for both containers and services, one cannot assume that serviceID will be a valid one. Hence, I think the code change `serviceID = ep.ID()` should be brought here if `serviceID == ""`.